modules/phase_field/src/base/PhaseFieldApp.C

```diff
@@ -161,6 +162,8 @@ PhaseFieldApp::registerApps()
   registerApp(PhaseFieldApp);
 }
 
+// External entry point for dynamic object registration
+extern "C" void PhaseFieldApp__registerObjects(Factory & factory) { PhaseFieldApp::registerObjects(factory); }
```

Comment:
New external entry point in the Application. @idaholab/moose-developers please take a look at this and comment. @friedmud mentioned having more granularity here to say that he only wanted to register certain object types from an application. That _might_ make sense to avoid a name collision for instance, maybe the Application that I want to register has a Convection Kernel but I'm only interested in some boundary condition that the other application has. I could potentially register just one type of system if I add another argument to this new entry point:

`std::vector<RegistrationType> moose_systems`. 
Is this something we want to support?